gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/translator/GoTranslateVisitor.java

```diff
@@ -57,8 +58,8 @@ public GoTranslateVisitor(final String graphTraversalSourceName) {
     public Void visitDateLiteral(final GremlinParser.DateLiteralContext ctx) {
         // child at 2 is the date argument to datetime() and comes enclosed in quotes
         final String dtString = ctx.getChild(2).getText();
-        final Date dt = DatetimeHelper.parse(removeFirstAndLastCharacters(dtString));
-        sb.append("time.UnixMilli(" + dt.getTime() + ")");
+        final OffsetDateTime dt = DatetimeHelper.parse(removeFirstAndLastCharacters(dtString));
+        sb.append("time.UnixMilli(" +  dt.toInstant().toEpochMilli() + ")");
```

Comment:
We may be able to preserve more precision as well as the timezone offset via https://pkg.go.dev/time#Date